app/views/landing_page/blocks/_main_navigation.html.erb

```diff
@@ -2,47 +2,26 @@
-    <button class="app-b-main-nav__button app-b-main-nav__button--no-js govuk-link govuk-link--no-underline" aria-expanded="false" aria-controls="app-b-main-nav__nav" type="button">Menu<span class="app-b-main-nav__icon" aria-hidden="true"></span></button>
-
-    <nav id="app-b-main-nav__nav" aria-label="Secondary">
-      <ul class="app-b-main-nav__ul">
-        <% contents_list(request.path, block.links[1..]).each do |link| %>
-          <%
-            li_classes = "app-b-main-nav__listitem"
-            li_classes << " app-b-main-nav__listitem--active" if link[:active]
-          %>
-          <%= tag.li(class: li_classes) do %>
-            <%
-              link_classes = "app-b-main-nav__link govuk-link govuk-link--no-underline govuk-link--no-visited-state"
-              link_aria = { current: true } if link[:active]
-            %>
-            <%= tag.a(href: link[:href], class: link_classes, aria: link_aria) do %>
-              <span class="app-b-main-nav__mobile-border"></span>
-              <%= link[:text] %>
-            <% end %>
-            <% if link[:links] %>
-              <ul class="app-b-main-nav__childlist">
-                <% link[:links].each do |child_link| %>
-                  <%
-                    child_link_classes = "app-b-main-nav__link govuk-link govuk-link--no-underline govuk-link--no-visited-state"
-                    child_link_aria = { current: true } if child_link[:active]
-                    child_li_classes = "app-b-main-nav__listitem"
-                    child_li_classes << " app-b-main-nav__listitem--active" if child_link[:active]
-                  %>
-                  <%= tag.li(class: child_li_classes) do %>
-                    <%= tag.a(href: child_link[:href], class: child_link_classes, aria: child_link_aria) do %>
-                      <span class="app-b-main-nav__mobile-border"></span>
-                      <%= child_link[:text] %>
-                    <% end %>
-                  <% end %>
-                <% end %>
-              </ul>
-            <% end %>
+  <h2 class="govuk-visually-hidden">Secondary navigation menu</h2>
+  <div class="app-b-main-nav__button-container app-b-main-nav__button-container--collapsed js-app-b-main-nav__button-container">
+    <div class="govuk-width-container">
+      <button type="button" class="app-b-main-nav__button app-b-main-nav__button--no-js govuk-link govuk-link--no-underline" aria-expanded="false" aria-controls="app-b-main-nav__nav-container">
+        Landing page menu
+      </button>
+    </div>
+  </div>
+  <div class="app-b-main-nav__nav-container app-b-main-nav__nav-container--js-hidden govuk-width-container js-app-b-main-nav__nav-container" id="app-b-main-nav__nav-container">
+    <nav aria-label="Secondary">
```

Comment:
I can see we've also got an visually hidden `h2` on line 5 which says "Secondary navigation menu" so do we need an `aria-label` here? Or perhaps we could add an `id` to the `h2` and reference it via `aria-labelledby`?

Reply:
Cool, I can use `aria-labelledby` as adding the label was recommended by our accessibility specialist.